js/ui/data_grid/ui.data_grid.editing.js

```diff
@@ -786,14 +785,17 @@ exports.EditingController = gridCore.ViewController.inherit((function() {
                 if(deferred) {
                     doneDeferred = $.Deferred();
                     deferred
-                        .always(function() { processedKeys.push(editData.key); })
+                        .always(function(data) {
+                            if(data === "cancel" && getEditMode(that) === DATAGRID_EDIT_MODE_CELL) {
+                                that._focusEditingCell();
```

Comment:
It is not a good place for the fix. Currently, all _focusEditingCell calls are in saveEditData, not in _saveEditDataCore. So, I think that this fix should be in saveEditData also.